src/workers/personalizedDigestEmail.ts

```diff
@@ -215,28 +274,21 @@ const worker: Worker = {
       return;
     }
 
-    const [dayName] = Object.entries(DayOfWeek).find(
-      ([, value]) => value === personalizedDigest.preferredDay,
-    );
-    const userName = user.name?.trim().split(' ')[0] || user.username;
-    const variationProps = await getEmailVariation(
-      personalizedDigest.variation,
-      userName,
-      dayName,
+    const variationProps = await getEmailVariation({
+      personalizedDigest,
       posts,
-    );
+      user,
+      feature: digestFeature,
+      currentDate,
+    });
     const emailPayload: MailDataRequired = {
       ...baseNotificationEmailData,
-      to: {
-        email: user.email,
-        name: userName,
-      },
       sendAt: Math.floor(emailSendDate.getTime() / 1000),
-      templateId: emailTemplateId,
+      templateId: digestFeature.templateId,
```

Comment:
Potentially fallback to `features.personalizedDigest.defaultValue.templateId` in case of partial object in default. Don't think growthbook deep merges default values.

Reply:
Good point about deep merge. Double check that. You can connect to the dev env of GB

Reply:
It does not, I utilized `deepmerge`, we already had it as peer dependency so just added it as depndency now.